components/core/SearchInput.tsx

```diff
@@ -19,26 +18,48 @@ const geocoder = new MapboxGeocoder({
 });
 
 function SearchInput(props: SearchInputProps): ReactElement {
-  const { placeholder, ariaLabel, onSubmit } = props;
+  const { placeholder, ariaLabel } = props;
   const router = useRouter();
+  const [searchInput, setSearchInput] = useState("");
+  const toast = useToast();
 
-  useEffect(() => {
-    const handleResult = (events: { result: Result }) => {
-      const { result } = events;
+  const handleResult = (events: { result: Result }) => {
+    const { result } = events;
 
-      router.push({
-        pathname: RoutePath.Search,
-        query: {
-          center: result.center,
-          place: result.place_name,
-          text: result.text,
-        },
-      });
-    };
+    toast.closeAll();
+
+    router.push({
+      pathname: RoutePath.Search,
+      query: {
+        center: result.center,
+        place: result.place_name,
+        text: result.text,
+      },
+    });
+  };
+
+  const handleError = () => {
+    toast({
+      id: "error",
+      title: "Location Not Found",
+      description: "Location entered couldn't be found, please try again.",
+      status: "error",
+      duration: 9000,
+      isClosable: true,
+    });
+  };
 
+  const onSubmit = () => {
+    geocoder.query(searchInput);
+    geocoder.on("result", handleResult);
```

Comment:
Not sure if `geocoder.on("result", handleResult)` is necessary here since the handler is already added in the `useEffect` hook. I think just `geocoder.query(searchInput)` by itself should work.

Nitpick: if that's the case, then `handleError` and `handleResult` could be moved in the `useEffect` hook to please the linter a bit :sweat_smile: 

![image](https://user-images.githubusercontent.com/20251243/134241910-5871bb55-e689-4995-aca4-4ded531823d3.png)

Reply:
fixed 63cc550